cloud/digitalocean/reconciler.go

```diff
@@ -213,10 +212,14 @@ func newClusterDefaults(base *cluster.Cluster) *cluster.Cluster {
 	return new
 }
 
-func handleCtrlC(c chan os.Signal) {
-	sig := <-c
-	if sig == syscall.SIGINT {
-		sigCaught = true
-		logger.Critical("Detected SIGINT. Please be patient while kubicorn cleanly exits. Maybe get a cup of tea?")
-	}
+func handleCtrlC(h *signals.Handler) {
+	go func() {
+		for {
+			if h.GetState() != 0 {
+				sigCaught = true
+				logger.Critical("Detected signal. Please be patient while kubicorn cleanly exits. Maybe get a cup of tea?")
```

Comment:
Can we get a `sync.Mutex` on this log line please? I notice it will pop up a lot..